src/core_modules/capture-core/components/Pages/ViewEvent/epics/viewEvent.epics.js

```diff
@@ -30,30 +32,29 @@ import { eventWorkingListsActionTypes } from '../../../WorkingLists/EventWorking
 import { resetLocationChange } from '../../../LockedSelector/QuickSelector/actions/QuickSelector.actions';
 import { buildUrlQueryString } from '../../../../utils/routing';
 
-export const getEventOpeningFromEventListEpic = (action$: InputObservable, store: ReduxStore) =>
+export const getEventOpeningFromEventListEpic = (action$: InputObservable) =>
     action$.pipe(
         ofType(eventWorkingListsActionTypes.VIEW_EVENT_PAGE_OPEN),
-        switchMap(({ payload: { eventId } }) => {
-            const state = store.value;
-            return getEvent(eventId)
-                .then((eventContainer) => {
-                    if (!eventContainer) {
-                        return openViewEventPageFailed(
-                            i18n.t('Event could not be loaded. Are you sure it exists?'));
-                    }
-                    const orgUnit = state.organisationUnits[eventContainer.event.orgUnitId];
-                    return startOpenEventForView(eventContainer, orgUnit);
-                })
-                .catch((error) => {
-                    const { message, details } = getErrorMessageAndDetails(error);
-                    log.error(
-                        errorCreator(
-                            message ||
-                            i18n.t('Event could not be loaded'))(details));
+        switchMap(({ payload: { eventId } }) => getEvent(eventId)
+            .then(eventContainer => (eventContainer ? Promise.all([eventContainer, getRulesEngineOrgUnit(eventContainer.event.orgUnitId)]) : []))
```

Comment:
The path to the rules engine starting with a click in a single event program's event list goes through here. The async method `getEvent` was already used here, so should be safe to fetch the orgUnit async at the same time. Note that `orgUnitId` is part of the result of the first async call.

Reply:
👍